### PR TITLE
[hmac/rtl] Fix idle_o when told to stop

### LINKED ISSUE #23036: [hmac/spec] Output pin doesn't go to Idle in case of HMAC stop invoked
### Description

Seen in simulation while testing the Save and Restore feature, the output pin `idle_o` is not moving into idle state after a hash_stop has been triggered.

### PR DESCRIPTION
**Pull request by andreaskurth**
Prior to this commit, `hmac_core` and thus the entire `hmac` module would not raise the `idle_o` output when it got told to stop (#23036). This is problematic because then HMAC cannot be clock-gated during context switch (save & restore feature).  Clock gating would only work when a message stream is fully processed (after the *process* command).

This commit fixes the problem by keeping track of idleness inside `hmac_core`.  In addition to the conditions that existed before, this tracker goes to idle when a *stop* command is received and the current message block is completely processed.

This fixes #23036.

### CODE REVIEW

**Comment by ballifatih** [comment]
LGTM, but probably best to wait for @gdessouky's approval. I am not super familiar with the internal logic.

This is also the Idle signal that could be reused for SW I think. https://github.com/lowRISC/opentitan/issues/22916

**Comment by andreaskurth** [comment]
> This is also the Idle signal that could be reused for SW I think.

Good idea; I'll mention it in #22916

**Review by martin-velay — APPROVED** [review]
LGTM, need to re-enable test as the piece of code from the SCB has been commented out with this tag `#23036`

**Comment by gdessouky** [comment]
Thanks, this LGTM, if it doesn't break existing DV. I have a lot of PRs waiting to merged that touch DV with fixes.

**Review by gdessouky — APPROVED** [review]

**Comment by andreaskurth** [comment]
The CW310 CI check failures seem to be unrelated to this PR; the other CI checks pass. Merging

**Comment by gdessouky** [comment]
> > This is also the Idle signal that could be reused for SW I think.
> 
> Good idea; I'll mention it in #22916

I think we should actually be using the `idle_o` from `hmac.sv` (see https://github.com/lowRISC/opentitan/pull/23242/commits) not `hmac_core.sv `that was modified here.

**Comment by andreaskurth** [comment]
> > > This is also the Idle signal that could be reused for SW I think.
> > 
> > 
> > Good idea; I'll mention it in #22916
> 
> I think we should actually be using the `idle_o` from `hmac.sv` (see https://github.com/lowRISC/opentitan/pull/23242/commits) not `hmac_core.sv `that was modified here.

Indeed :+1:

**Pull request merged**
